gateleen-cache/src/main/java/org/swisspush/gateleen/cache/fetch/DefaultCacheDataFetcher.java

```diff
@@ -31,43 +31,42 @@ public DefaultCacheDataFetcher(HttpClient httpClient) {
 
-            cRes.bodyHandler(data -> {
-                if (StatusCode.OK.getStatusCode() == cRes.statusCode()) {
+                        String contentType = cRes.getHeader(CONTENT_TYPE_HEADER);
+                        if(contentType != null && !contentType.contains(CONTENT_TYPE_JSON)){
+                            log.warn("Content-Type {} is not supported", contentType);
+                            future.complete(Result.err(StatusCode.UNSUPPORTED_MEDIA_TYPE));
+                            return;
+                        }
 
-                    String contentType = cRes.getHeader(CONTENT_TYPE_HEADER);
-                    if(contentType != null && !contentType.contains(CONTENT_TYPE_JSON)){
-                        log.warn("Content-Type {} is not supported", contentType);
-                        future.complete(Result.err(StatusCode.UNSUPPORTED_MEDIA_TYPE));
-                        return;
+                        future.complete(Result.ok(data));
+                    } else {
+                        StatusCode statusCode = StatusCode.fromCode(cRes.statusCode());
+                        if(statusCode == null) {
+                            log.error("Got unknown status code {} while fetching cache data. Using 500 Internal Server Error " +
+                                    "instead", cRes.statusCode());
+                            statusCode = StatusCode.INTERNAL_SERVER_ERROR;
+                        }
+                        future.complete(Result.err(statusCode));
                     }
-
-                    future.complete(Result.ok(data));
-                } else {
-                    StatusCode statusCode = StatusCode.fromCode(cRes.statusCode());
-                    if(statusCode == null) {
-                        log.error("Got unkown status code {} while fetching cache data. Using 500 Internal Server Error " +
-                                "instead", cRes.statusCode());
-                        statusCode = StatusCode.INTERNAL_SERVER_ERROR;
-                    }
-                    future.complete(Result.err(statusCode));
+                }),
+                event -> {
+                    log.warn("Got an error while fetching cache data", event);
+                    future.complete(Result.err(StatusCode.INTERNAL_SERVER_ERROR));
                 }
-            });
-
-            cRes.exceptionHandler(event -> {
-                log.warn("Got an error while fetching cache data", event);
-                future.complete(Result.err(StatusCode.INTERNAL_SERVER_ERROR));
-            });
-
-        });
+        );
 
-        cReq.setTimeout(requestTimeoutMs);
-        cReq.headers().setAll(requestHeaders);
-        cReq.headers().set("Accept", "application/json");
-        cReq.headers().set(SELF_REQUEST_HEADER, "true");
-        cReq.setChunked(true);
-        cReq.end();
+        fetchRequest.putHeader("Accept", "application/json");
+        fetchRequest.putHeader(SELF_REQUEST_HEADER, "true");
```

Comment:
Any reason to set those headers after creating the request? Looks as the request creation method already takes headers. Are those the same?

Reply:
Yes those are the same headers. There is no particular reason to not already add those headers to the exsiting MultiMap. In this case, the intention was to show that those are additional headers for the client request.